syntaxes/GDShader.tmLanguage.json

```diff
@@ -314,51 +314,65 @@
         },
         "literalFloat": {
             "name": "constant.numeric.float.gdshader",
-            "match": "\b(?:\\d+[eE][-+]?\\d+|(?:\\d*[.]\\d+|\\d+[.])(?:[eE][-+]?\\d+)?)[fF]?"
+            "match": "\\b(?:\\d+[eE][-+]?\\d+|(?:\\d*[.]\\d+|\\d+[.])(?:[eE][-+]?\\d+)?)[fF]?"
         },
         "literalInt": {
             "name": "constant.numeric.integer.gdshader",
-            "match": "\b(?:0[xX][0-9A-Fa-f]+|\\d+[uU]?)\b"
+            "match": "\\b(?:0[xX][0-9A-Fa-f]+|\\d+[uU]?)\\b"
         },
         "literalBool": {
             "name": "constant.language.boolean.gdshader",
-            "match": "\b(?:false|true)\b"
+            "match": "\\b(?:false|true)\\b"
         },
         "identifierType": {
             "name": "entity.name.type.gdshader",
-            "match": "\b[a-zA-Z_]\\w*(?=(?:\\s*\\[\\s*\\w*\\s*\\])?\\s+[a-zA-Z_]\\w*\b)"
+            "match": "\\b[a-zA-Z_]\\w*(?=(?:\\s*\\[\\s*\\w*\\s*\\])?\\s+[a-zA-Z_]\\w*\\b)"
         },
         "constructor": {
             "name": "entity.name.type.constructor.gdshader",
-            "match": "\b[a-zA-Z_]\\w*(?=\\s*\\[\\s*\\w*\\s*\\]\\s*[(])|\b[A-Z]\\w*(?=\\s*[(])"
+            "match": "\\b[a-zA-Z_]\\w*(?=\\s*\\[\\s*\\w*\\s*\\]\\s*[(])|\\b[A-Z]\\w*(?=\\s*[(])"
         },
         "processorFunction": {
             "name": "support.function.gdshader",
-            "match": "\b(?:vertex|fragment|light|start|process|sky|fog)(?=(?:\\s|/\\*(?:\\*(?!/)|[^*])*\\*/)*[(])"
+            "match": "\\b(?:vertex|fragment|light|start|process|sky|fog)(?=(?:\\s|/\\*(?:\\*(?!/)|[^*])*\\*/)*[(])"
         },
         "identifierFunction": {
             "name": "entity.name.function.gdshader",
-            "match": "\b[a-zA-Z_]\\w*(?=(?:\\s|/\\*(?:\\*(?!/)|[^*])*\\*/)*[(])"
+            "match": "\\b[a-zA-Z_]\\w*(?=(?:\\s|/\\*(?:\\*(?!/)|[^*])*\\*/)*[(])"
         },
         "swizzling": {
-            "name": "variable.other.property.gdshader",
-            "match": "(?<=[.]\\s*)(?:[xyzw]{2,4}|[rgba]{2,4}|[stpq]{2,4})\b"
+            "match": "([.])\\s*([xyzw]{2,4}|[rgba]{2,4}|[stpq]{2,4})\\b",
+            "captures": {
+                "1": {
+                    "name": "punctuation.accessor.gdshader"
+                },
+                "2": {
+                    "name": "variable.other.property.gdshader"
+                }
+            }
         },
         "identifierField": {
-            "name": "entity.name.variable.field.gdshader",
-            "match": "(?<=[.]\\s*)[a-zA-Z_]\\w*\b"
+            "match": "([.])\\s*([a-zA-Z_]+)\\w*\\b(?!\\()",
```

Comment:
Don't forget to capture the `\\w*`, like I said before; and remove the `+`:
"...`([a-zA-Z_]\\w*)`..."

Also, this case won't catch `.length ()` with a space (some people might use this kind of weird code style?). This can be solved with a `\\s*`, but I'm not familiar with linguist, so I don't know if you can use variable-length negative lookahead (presumably yes?). If you can't, maybe adding it right before the lookahead should be enough.

Reply:
>  not familiar with linguist, so I don't know if you can use variable-length negative lookahead (presumably yes?). If you can't, maybe adding it right before the lookahead should be enough.

I'll try and run both versions against their "compiler", just for the sake of education.

Reply:
Linguist's compiler did not complain about variable-length negative lookahead. Some reading I did earlier implied that despite being syntactically similar, lookahead and lookbehind are implemented dramatically differently in the regex state machine, with lookbehind being a much more complicated process. 

> `.length ()` with a space (some people might use this kind of weird code style?)

As much as I don't want to encourage this kind of behavior, this rule correctly supports whatever mangled whitespace somebody wants to insert. I'm gonna have to review my gdscript grammar and make sure it's similarly flexible.

Thanks for the regex review, I feel like I picked up some new tricks!

Reply:
> I think the negative-lookahead also has stronger guarantees about not "consuming" what it matches against?

Yes! In VSCode grammar, lookaheads/behinds don't consume the input. At least the `\\(` has to be inside a lookahead; the `\\s*` could be outside it if needed (e.g. `\\b\\s*(?!\\()`), but since you say it'll likely work, then (B) is indeed better. (A) seems wrong, because `^` is start of line and also like I said `(` should not be consumed as part of the token.